src/tools/r2rdump/EHInfo.cs

```diff
@@ -98,21 +113,29 @@ public void WriteTo(StringBuilder stringBuilder, int methodRva)
             stringBuilder.Append($@"HndEnd {HandlerEnd:X4} (RVA {(HandlerEnd + methodRva):X4}) ");
             stringBuilder.Append($@"ClsFlt {ClassTokenOrFilterOffset:X4}");
 
-            if ((Flags & CorExceptionFlag.COR_ILEXCEPTION_CLAUSE_FILTER) != CorExceptionFlag.COR_ILEXCEPTION_CLAUSE_NONE)
+            switch (Flags & CorExceptionFlag.COR_ILEXCEPTION_CLAUSE_KIND_MASK)
             {
-                stringBuilder.Append($@" (RVA {(ClassTokenOrFilterOffset + methodRva):X4})");
+                case CorExceptionFlag.COR_ILEXCEPTION_CLAUSE_NONE:
+                    stringBuilder.AppendFormat(" CATCH: {0}", ClassName ?? "null");
+                    break;
 
-                stringBuilder.Append(" FILTER");
-            }
-            if ((Flags & CorExceptionFlag.COR_ILEXCEPTION_CLAUSE_FINALLY) != CorExceptionFlag.COR_ILEXCEPTION_CLAUSE_NONE)
-            {
-                stringBuilder.Append(" FINALLY");
-            }
-            if ((Flags & CorExceptionFlag.COR_ILEXCEPTION_CLAUSE_FAULT) != CorExceptionFlag.COR_ILEXCEPTION_CLAUSE_NONE)
-            {
-                stringBuilder.Append(" FAULT");
+                case CorExceptionFlag.COR_ILEXCEPTION_CLAUSE_FILTER:
+                    stringBuilder.AppendFormat(" FILTER (RVA {0:X4})", ClassTokenOrFilterOffset + methodRva);
+                    break;
+
+                case CorExceptionFlag.COR_ILEXCEPTION_CLAUSE_FINALLY:
+                    stringBuilder.AppendFormat(" FINALLY (RVA {0:X4})", ClassTokenOrFilterOffset + methodRva);
```

Comment:
In the FINALLY and FAULT cases, the "ClassToken"/"FilterOffset" field is unused, so I don't think you should try to interpret it.